Marlin/src/lcd/menu/menu_configuration.cpp

```diff
@@ -119,19 +119,37 @@ static void lcd_factory_settings() {
 
 #endif
 
-#if ENABLED(DUAL_X_CARRIAGE)
-
+#if HAS_HOTEND_OFFSET
   #include "../../module/motion.h"
   #include "../../gcode/queue.h"
 
-  void _recalc_IDEX_settings() {
-    if (active_extruder) {                      // For the 2nd extruder re-home so the next tool-change gets the new offsets.
+  void _recalc_offsets() {
+    if (active_extruder && all_axes_known()) {  // For the 2nd extruder re-home so the next tool-change gets the new offsets.
       enqueue_and_echo_commands_P(PSTR("G28")); // In future, we can babystep the 2nd extruder (if active), making homing unnecessary.
       active_extruder = 0;
```

Comment:
Is this the behavior we want for all tools that have hotend offsets? I think not! Only Dual X uses the `hotend_offset` for X in a special way that might require homing. All other types of tools (and Dual X too) can and should recalculate the current position, if it is needed. Most of the time the current position will not be affected. Since the `hotend_offset` is only applied on probing, homing-with-probe, and tool-change, the updated values will simply take effect on the next one of those operations.

Reply:
If active_extruder !=0 and these numbers are changed we would need to know old and new to recalculate the position. Do we add the extra variable in, do we hide this menu when T0 isnt active, then no adjustment / recalc is needed, or simply allow homing to accomplish it as was done for IDEX? IDEX does handle X specially however the position will be modified and incorrect on anything with an offset. The issue will come in on the next tool change as the return will use these new values and position wont be whats expected.

Reply:
Hotend offsets can be modified anytime outside of a print job. Modifying them should ideally update the current position as-needed, and nothing more. The current position does not necessarily need to change if the second tool is selected. It may already properly reflect its physical position in the coordinate space. But, assuming the second tool was selected while the hotend offsets were incorrect, all you need to do is apply the change in tool offset to the position. If mesh leveling happens to be enabled, then the Z position will also need to be corrected.

Reply:
Note that the `hotend_offset` for X on a dual-x machine is not an offset, but the home position of the second tool.

Reply:
For some types of menu-editable values, it will simplify things to make a wrapper class or struct and then require using the wrapper class for editable values. This will help standardize behavior and allow better handling of cases where the old and new values (or other things) need to be used in some logical way in value editing. A basic getter-setter pattern is probably all that's needed.